pumpkin-protocol/src/server/handshake/mod.rs

```diff
@@ -1,25 +1,13 @@
 use pumpkin_macros::packet;
+use serde::Deserialize;
 
-use crate::{
-    bytebuf::{ByteBuffer, DeserializerError},
-    ConnectionState, ServerPacket, VarInt,
-};
+use crate::{ConnectionState, VarInt};
 
 #[packet(0x00)]
+#[derive(Deserialize)]
 pub struct SHandShake {
     pub protocol_version: VarInt,
     pub server_address: String, // 255
     pub server_port: u16,
     pub next_state: ConnectionState,
 }
```

Comment:
Hey, The only reason i used manuel Deserialization here is because the server address has a max length of 255. There where plans of implementing maybe some kinda struct called `BoundedString`, Not sure tough how

Reply:
I've been playing with a solution to this problem for fixed length bitsets or similar, but not addressed it in depth.

In theory, it's possible to make a newtype wrapper like
```rust
struct BoundedLengthString<const LEN: usize>(String);
```

and implement a string visitor that restricts the size, implementing `From`/`Into` would allow you to use 
```
#[serde(try_from = BoundedLengthString<255>)]
pub server_address: String
```

but there are some considerations to make regarding strings in general. The current implementation is actually incorrect in that it assumes string length is in bytes. However, according to the wiki Strings are actually prefixed with their length in *characters*, making the implementation faulty on Non-ASCII/english characters.

Reply:
@Snowiiii Imo @StripedMonkey 's solution is cleaner, since we dont need to cast to a different type when using the packet.

---

@StripedMonkey 
> but there are some considerations to make regarding strings in general. The current implementation is actually incorrect in that it assumes string length is in bytes. However, according to the wiki Strings are actually prefixed with their length in characters, making the implementation faulty on Non-ASCII/english characters.

^^ regarding this, i dont think thats true tho, on wiki.vg it says that a string is a `UTF-8 string prefixed with its size in bytes as a VarInt.`, the size here does not refer to amount of characters, but amount of bytes. This can be easily checked, by simply sending a chat message in pumpkin with the current implementation, with characters outside of the single byte range of UTF-8
![image](https://github.com/user-attachments/assets/0ab7b385-d8dd-475d-a427-3f70ca72499d)

Reply:
Sorry, I missread, Max string length in characters is what I was actually reading. So the limits provided in the wki aren't in bytes.